### PR TITLE
Quote connection impersonation roles

### LINKED ISSUE #33364: Role-based security at database level does not work if the role contains special characters such as the dot. (v0.47)
### Describe the bug

Role-based security at database level does not work if the role contains special characters such as the dot.
I think I'm missing the double quote on the role during the sql query.
https://www.postgresql.org/docs/15/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS

I use the login from the ldap attribute as the role for postgresql (which is made up of the first name, a dot and the last name)

example login/role
"homer.simpson"

### To Reproduce

configur user attribut (login)
impersonated

error with user "homer.simpson"
error

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
metabase : version 0.47

postgresql v15
```

### Severity

blocking new feature security

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by noahmoss**
Resolves https://github.com/metabase/metabase/issues/33364

Quotes roles in `SET ROLE`/`USE ROLE` commands for postgres & snowflake, respectively. Updates tests to use role names with a `.` in them, similar to the report in the issue.

Omits quote the `SET ROLE NONE` command to reset the role for postgres. It seems to work with quotes on my local postgres instance but causes a very confusing exception in tests. I think it's triggering a bug deep in the test infra somehow which I want to figure out, but don't want it to block this PR since it's fixing a release blocker.

```
ERROR in metabase-enterprise.advanced-permissions.driver.impersonation-test during :clojure.test/each-fixtures (core_deftype.clj:584)
Uncaught exception during fixture initialization.

java.lang.IllegalArgumentException: No implementation of method: :explain of protocol: #'schema.core/Schema found for class: clojure.lang.Var$Unbound
                          clojure.core/-cache-protocol-fn  core_deftype.clj:  584
```

### CODE REVIEW

File: src/metabase/driver/postgres.clj [hunk 1]
```diff
@@ -817,7 +817,9 @@
 
 (defmethod driver.sql/set-role-statement :postgres
   [_ role]
-  (format "SET ROLE %s;" role))
+  (if (= (u/upper-case-en role) "NONE")
+   (format "SET ROLE %s;" role)
+   (format "SET ROLE \"%s\";" role)))
```
Comment (metamben) [inline src/metabase/driver/postgres.clj:822]: Can `role` contain `"` characters? Can it be escaped?

**Review by metamben — APPROVED** [review]
Approving with the assumption that `role` strings cannot contain quotes.

File: modules/drivers/snowflake/src/metabase/driver/snowflake.clj [hunk 2]
```diff
@@ -583,7 +583,7 @@
 
 (defmethod driver.sql/set-role-statement :snowflake
   [_ role]
-  (format "USE ROLE %s;" role))
+  (format "USE ROLE \"%s\";" role))
```
Comment (metamben) [inline modules/drivers/snowflake/src/metabase/driver/snowflake.clj:586]: Can `role` contain `"` characters? Can it be escaped?

File: src/metabase/driver/postgres.clj (hunk 1 above)
Comment (noahmoss) [reply src/metabase/driver/postgres.clj:822]: Tested on both postgres and snowflake, can't seem to manually create a role that includes `"`, even by escaping them

**Pull request merged**